workflow/validate/validate.go

```diff
@@ -205,7 +205,8 @@ func ValidateWorkflow(wftmplGetter templateresolution.WorkflowTemplateNamespaced
 		}
 	}
 	if wf.Spec.OnExit != "" {
-		// now when validating onExit, {{workflow.status}} is now available as a global
+		// now when validating onExit, {{workflow.status}} and {{workflow.duration}} is now available as a global
+		ctx.globalParams[common.GlobalVarWorkflowDuration] = placeholderGenerator.NextPlaceholder()
```

Comment:
test?

Reply:
Thanks for your quick response! This is still a bit in-progress, because I'm not really sure if adding this global var here causes any other problems. I've also created an issue for this: https://github.com/argoproj/argo-workflows/issues/5484

I will add a test soon.